code/modules/fishing/sources/_fish_source.dm

```diff
@@ -123,31 +112,34 @@ GLOBAL_LIST_INIT(preset_fish_sources,init_fishing_configurations())
 	source.used_rod?.consume_bait(reward)
 
 /// Gives out the reward if possible
-/datum/fish_source/proc/dispense_reward(reward_path, mob/fisherman, fishing_spot)
+/datum/fish_source/proc/dispense_reward(reward_path, mob/fisherman, turf/fishing_spot)
 	if((reward_path in fish_counts)) // This is limited count result
-		if(fish_counts[reward_path] > 0)
-			fish_counts[reward_path] -= 1
-		else
-			reward_path = FISHING_DUD //Ran out of these since rolling (multiple fishermen on same source most likely)
-	var/atom/movable/reward
-	if(ispath(reward_path))
-		reward = new reward_path(get_turf(fisherman))
-		if(ispath(reward_path,/obj/item))
-			if(ispath(reward_path,/obj/item/fish))
-				var/obj/item/fish/caught_fish = reward
-				caught_fish.randomize_size_and_weight()
-				//fish caught signal if needed goes here and/or fishing achievements
-			//Try to put it in hand
-			INVOKE_ASYNC(fisherman, TYPE_PROC_REF(/mob, put_in_hands), reward)
-			fisherman.balloon_alert(fisherman, "caught [reward]!")
-		else //If someone adds fishing out carp/chests/singularities or whatever just plop it down on the fisher's turf
-			fisherman.balloon_alert(fisherman, "caught something!")
-	else if (reward_path == FISHING_DUD)
-		//baloon alert instead
+		fish_counts[reward_path] -= 1
+		if(!fish_counts[reward_path])
+			fish_counts -= reward_path //Ran out of these since rolling (multiple fishermen on same source most likely)�
+
+	var/atom/movable/reward = spawn_reward(reward_path, fisherman, fishing_spot)
+	if(!reward) //baloon alert instead
 		fisherman.balloon_alert(fisherman,pick(duds))
+		return
+	if(isitem(reward)) //Try to put it in hand
+		INVOKE_ASYNC(fisherman, TYPE_PROC_REF(/mob, put_in_hands), reward)
+	fisherman.balloon_alert(fisherman, "caught [reward]!")
 	SEND_SIGNAL(fisherman, COMSIG_MOB_FISHING_REWARD_DISPENSED, reward)
-	if(reward)
-		SEND_SIGNAL(reward, COMSIG_ATOM_FISHING_REWARD, fishing_spot)
+	return reward
+
+/// Spawns a reward from a atom path right where the fisherman is. Part of the dispense_reward() logic.
+/datum/fish_source/proc/spawn_reward(reward_path, mob/fisherman,  turf/fishing_spot)
+	if(reward_path == FISHING_DUD)
+		return
+	if(ispath(reward_path, /datum/chasm_detritus))
+		return GLOB.chasm_detritus_types[reward_path].dispense_reward(fishing_spot)
```

Comment:
I don't love that we're typechecking for one specific type of item here, but probably it's premature to make a generic `fishing_reward` datum before more than one thing uses it.